### ISSUE TITLE
typescript mode does not recognize /** @type {{[key: string]: string}} */

### ISSUE THREAD

**Issue by thernstig** [gajus/eslint-plugin-jsdoc#796]
## Expected behavior

This should be valid (it is called an index signature https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures).
```js
/** @type {{[key: string]: string}} */
const obj = {};
obj.a = 'foo';
obj.b = 5;
```

## Actual behavior
It is not as it gives the error `Syntax error in type: {[key: string]: string}`

## ESLint Config

```json
  "settings": {
    "jsdoc": {
      "mode": "typescript",
    }
  }
```

I am unsure if this should have been written at https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues, because the repo hardly has any activity at all.

## Environment

- Node version: v14.17.6
- ESLint version: v7.32.0
- `eslint-plugin-jsdoc` version: 36.1.0

**Comment by brettz9** [comment]
`jsdoc-type-pratt-parser` has had activity, and this should indeed be filed there (though I suspect it is a dupe of https://github.com/jsdoc-type-pratt-parser/jsdoc-type-pratt-parser/issues/96 ).

Closing, but feel free to comment further as needed.
